server/valgrind-out.txt

```diff
@@ -1,144 +1,134 @@
+--21905-- REDIR: 0x4920850 (libc.so.6:strcasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x49203d0 (libc.so.6:memchr) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x493c930 (libc.so.6:wcslen) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x491fae0 (libc.so.6:strspn) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x4920750 (libc.so.6:stpncpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x49206f0 (libc.so.6:stpcpy) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x4921c90 (libc.so.6:strchrnul) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x49208a0 (libc.so.6:strncasecmp_l) redirected to 0x48311d0 (_vgnU_ifunc_wrapper)
+--21905-- REDIR: 0x4a08500 (libc.so.6:__strrchr_avx2) redirected to 0x483ea10 (rindex)
+--21905-- REDIR: 0x4a0bb60 (libc.so.6:__memset_avx2_unaligned_erms) redirected to 0x48428e0 (memset)
+--21905-- REDIR: 0x491a0e0 (libc.so.6:malloc) redirected to 0x483b780 (malloc)
+--21905-- REDIR: 0x491a6d0 (libc.so.6:free) redirected to 0x483c9d0 (free)
+--21905-- REDIR: 0x4a08310 (libc.so.6:__strchrnul_avx2) redirected to 0x4843540 (strchrnul)
+--21905-- REDIR: 0x4a09c10 (libc.so.6:__strcpy_avx2) redirected to 0x483f090 (strcpy)
+--21905-- REDIR: 0x491bb10 (libc.so.6:calloc) redirected to 0x483dce0 (calloc)
+--21905-- REDIR: 0x4a086d0 (libc.so.6:__strlen_avx2) redirected to 0x483ef40 (strlen)
+--21905-- REDIR: 0x4a0b6e0 (libc.so.6:__memcpy_avx_unaligned_erms) redirected to 0x48429f0 (memmove)
+--21905-- REDIR: 0x491ffa0 (libc.so.6:__GI_strstr) redirected to 0x4843ca0 (__strstr_sse2)
+--21905-- REDIR: 0x4a04cc0 (libc.so.6:__memcmp_avx2_movbe) redirected to 0x48421e0 (bcmp)
+--21905-- REDIR: 0x4a03bd0 (libc.so.6:__strcmp_avx2) redirected to 0x483fed0 (strcmp)
+--21905-- REDIR: 0x4a0b6c0 (libc.so.6:__mempcpy_avx_unaligned_erms) redirected to 0x4843660 (mempcpy)
+--21905-- REDIR: 0x491ae80 (libc.so.6:realloc) redirected to 0x483df30 (realloc)
+--21905-- REDIR: 0x4a080e0 (libc.so.6:__strchr_avx2) redirected to 0x483ebf0 (index)
+--21905-- Reading syms from /usr/lib/x86_64-linux-gnu/libgcc_s.so.1
+--21905--    object doesn't have a symbol table
+--21905-- REDIR: 0x4a0a900 (libc.so.6:__stpcpy_avx2) redirected to 0x4842300 (stpcpy)
+--21905-- Discarding syms at 0x56795e0-0x568a045 in /usr/lib/x86_64-linux-gnu/libgcc_s.so.1 (have_dinfo 1)
+==21905== 
+==21905== HEAP SUMMARY:
+==21905==     in use at exit: 680 bytes in 17 blocks
+==21905==   total heap usage: 535 allocs, 518 frees, 1,467,363 bytes allocated
+==21905== 
+==21905== Searching for pointers to 17 not-freed blocks
+==21905== Checked 91,336 bytes
+==21905== 
+==21905== 680 bytes in 17 blocks are still reachable in loss record 1 of 1
+==21905==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
+==21905==    by 0x10B30B: main (aesdsocket.c:419)
```

Comment:
@Ganeshkm10 this is probably highlighting a real problem, it seams your list free isn't actually happening.  My guess is you might be exiting early at https://github.com/cu-ecen-aeld/assignments-3-and-later-Ganeshkm10/pull/4/files#diff-05d8ef2740ea224be3f6af4411fca3542fe9e4e066c2d8a4b84b5190787125e0R405 ?  Did you see the print messages stating threads are being cancelled?